src/tds/codec/bulk_load.rs

```diff
@@ -104,4 +92,23 @@ where
 
         ExecuteResult::new(self.connection).await
     }
+    
+    async fn write_packets(&mut self) -> crate::Result<()> {
+        let packet_size = (self.connection.context().packet_size() as usize) - HEADER_BYTES;
+        
+        while self.buf.len() > packet_size {
```

Comment:
Condition changed to strictly greater intentionally, to avoid writing empty packet with EOM header, instead of including EOM header in the last "full" packet.